### PR TITLE
pkg/profiler: hack to fix double-registration  panic

### LINKED ISSUE #255: Duplicate metrics collector registration attempted
```
panic: duplicate metrics collector registration attempted

goroutine 68 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc000568f70, {0xc000568fb0, 0x1, 0x0})
    github.com/prometheus/client_golang@v1.12.1/prometheus/registry.go:403 +0x7f
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounterVec({{0x1d9a6e8, 0xc00022d590}}, {{0x0, 0x0}, {0x0, 0x0}, {0x1b77405, 0x29}, {0x1b65bd8, 0x20}, ...}, ...)
    github.com/prometheus/client_golang@v1.12.1/prometheus/promauto/auto.go:276 +0x133
github.com/parca-dev/parca-agent/pkg/profiler.NewCgroupProfiler({0x1d78d20, 0xc00022d4f0}, {0x1d9a6e8, 0xc00022d590}, 0xc000307740, {0x1d7a1e0, 0xc00023bf40}, {0x1d94610, 0xc000569bf0}, 0xc0005a1230, ...)
    github.com/parca-dev/parca-agent/pkg/profiler/profiler.go:103 +0x465
github.com/parca-dev/parca-agent/pkg/target.(*ProfilerPool).Sync(0xc000566ea0, {0xc000648000, 0x9, 0x3})
    github.com/parca-dev/parca-agent/pkg/target/profiler_pool.go:132 +0x266
github.com/parca-dev/parca-agent/pkg/target.(*Manager).reconcileTargets(0xc000648e00, {0x1dadc70, 0xc000591000}, 0x0)
    github.com/parca-dev/parca-agent/pkg/target/manager.go:97 +0x1eb
github.com/parca-dev/parca-agent/pkg/target.(*Manager).Run(0xc000308dc0, {0x1dadc70, 0xc000591000}, 0xc0002d0660)
    github.com/parca-dev/parca-agent/pkg/target/manager.go:75 +0xc9
main.main.func8()
    github.com/parca-dev/parca-agent/cmd/parca-agent/main.go:370 +0x105
github.com/oklog/run.(*Group).Run.func1({0xc000591040, 0xc0001bca20})
    github.com/oklog/run@v1.1.0/group.go:38 +0x2f
created by github.com/oklog/run.(*Group).Run
    github.com/oklog/run@v1.1.0/group.go:37 +0x22f
[event: pod parca/parca-agent-4h6gm] Back-off restarting failed container
```

cc @Sylfrena @metalmatze

### PR DESCRIPTION
**Pull request by Sylfrena**
Signed-off-by: Sumera Priyadarsini <sylphrenadin@gmail.com>

Fix #255

### CODE REVIEW

File: cmd/parca-agent/main.go [hunk 1]
```diff
@@ -364,9 +365,15 @@ func main() {
 	// Run group for target manager
 	{
 		ctx, cancel := context.WithCancel(ctx)
+		counter := promauto.With(reg).NewCounterVec(
```
Comment (kakkoyun) [inline cmd/parca-agent/main.go:368]: Let's think a little ahead and create a `metrics` struct here to collect similar metrics.

File: cmd/parca-agent/main.go [hunk 2]
```diff
@@ -364,9 +365,15 @@ func main() {
...
+			prometheus.CounterOpts{
+				Name: "parca_agent_profiler_missing_stacks_total",
+				Help: "Number of missing profile stacks",
+			},
+			[]string{"type"})
 		g.Add(func() error {
 			level.Debug(logger).Log("msg", "starting target manager")
-			return tm.Run(ctx, m.SyncCh())
+			return tm.Run(ctx, m.SyncCh(), counter)
```
Comment (kakkoyun) [inline cmd/parca-agent/main.go:376]: I would rather see these types of dependencies pass through constructors and be an instance member of the type.

**Review by kakkoyun — COMMENTED** [review]
Good catch :+1: It's going in the right direction.

File: cmd/parca-agent/main.go (hunk 2 above)
Comment (Sylfrena) [reply cmd/parca-agent/main.go:376]: agreed, this was entirely a hack to  see if it works, if the approach seems correct, there's some refactoring and cleanup that needs to be done.

**Review by kakkoyun — COMMENTED** [review]
When that re-review this, after @metalmatze and I talked, I think we shouldn't pass the counter while we already passing the reg. We just need to add a label.

**Review by kakkoyun — APPROVED** [review]
LGTM

**Pull request merged**